UIforETW/Utility.cpp

```diff
@@ -349,78 +558,83 @@ int DeleteFiles(HWND hwnd, const std::vector<std::wstring>& paths)
 		FO_DELETE,
 		&fileNames[0],
 		NULL,
-		FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION,
+		(FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION),
 	};
 	// Delete using the recycle bin.
-	int result = SHFileOperation(&fileOp);
-
-	return result;
+	//TODO: IFileOperation?
+	return ::SHFileOperationW(&fileOp);
 }
 
 void SetClipboardText(const std::wstring& text)
 {
-	BOOL cb = OpenClipboard(GetDesktopWindow());
-	if (!cb)
+	ATLVERIFY(::OpenClipboard(::GetDesktopWindow()));

```

Comment:
That's fair, I was mainly just tossing it out as a note that GlobalAlloc failing wasn't necessarily in the same category as malloc failing.

The limit on count was found by experimentation. I first encountered it on XP or Vista at least and it is still there as of Win10. https://msdn.microsoft.com/en-us/library/ms810603.aspx <-- That references the limit, but it is an archived and ancient source. "If you work out the math you can see that the number of memory handles available for MOVEABLE memory is limited. 512K bytes / 8 bytes per handle = 65,535 handles. In fact, the system imposes this limitation on each process. Fortunately, this only applies to MOVEABLE memory."